effectful-core/src/Effectful/Internal/Monad.hs

```diff
@@ -40,15 +41,20 @@ module Effectful.Internal.Monad
 
   -- * Effects
 
+  -- * Dispatch
+  , Dispatch(..)
+  , DispatchOf
+  , Rep
```

Comment:
I have concerns about exporting this name. It clashes with `Rep` from `GHC.Generic`. If possible, I think we should choose a different (more descriptive) name.

Reply:
This is an internal module, it's not exported from any public one.